src/handler.js

```diff
@@ -80,6 +80,13 @@ const createBadRequestDetails = (event) => {
   return { code: httpStatus.BAD_REQUEST, message: `Your request cannot be processed because the supplied parameter(s) ${event.queryStringParameters} cannot be understood` }
 }
 
+const createErrorDetails = (event) => isGetMethod(event) ? problemsWithGetMethod(event) : createMethodNotAllowedDetails(event)
+
 const problemsWithGetMethod = event => hasQueryParameters(event) ? createBadRequestDetails(event) : createNotFoundDetails(event)
 
-const createErrorDetails = (event) => isGetMethod(event) ? problemsWithGetMethod(event) : createMethodNotAllowedDetails(event)
+const hasValidQueryParameters = (event) => event.queryStringParameters.query !== undefined && hasValidParameterNames(event)
+
+const hasValidParameterNames = (event) => {
+  const parameterNames = ['query', 'year', 'start']
+  return Object.keys(event.queryStringParameters).filter(key => (!parameterNames.includes(key))).length === 0
+}
```

Comment:
While the code works perfectly, I wonder if we should view query as a standard case, e.g.
```
const hasValidQueryParameters = (event) => {
  const querySpec = [{ name: 'query', required: true }, { name: 'year', required: false }, { name: 'start', required: false }]
  const queryStringKeys = Object.keys(event.queryStringParameters);
  return isValidParameterName(event, queryStringKeys) && hasRequiredParameters(querySpec, queryStringKeys)
}

const isValidParameterName = (actualKeys, querySpec) => actualKeys.every(key => querySpec.map(param => param.name).includes(key))

const hasRequiredParameters = (actualKeys, querySpec) => {
  const required = querySpec.filter(param => param.required === true).map(param => param.name)
  return required.every(item => actualKeys.includes(item))
}
```
This can later be refactored to a generic config in an external class